site/components/common/SEO/SEO.tsx

```diff
@@ -0,0 +1,158 @@
+import Head from 'next/head'
+import { FC, Fragment, ReactNode } from 'react'
+import config from '@config/seo_meta.json'
+
+const storeBaseUrl =
+  process.env.NEXT_PUBLIC_STORE_URL ?? process.env.VERCEL_URL
+    ? `https://${process.env.VERCEL_URL}`
+    : null
```

Comment:
I don't think this is going to work, `VERCEL_URL` won't be replaced by Next.js at build time. It needs to be added to `next.config.js` so that it gets replaced like variables that start with `NEXT_PUBLIC_`